pytorch_lightning/utilities/cli.py

```diff
@@ -889,9 +891,13 @@ def instantiate_class(args: Union[Any, Tuple[Any, ...]], init: Dict[str, Any]) -
 
 
 def _get_short_description(component: object) -> Optional[str]:
-    parse, _ = import_docstring_parse("LightningCLI(run=True)")
-    try:
-        docstring = parse(component.__doc__)
-        return docstring.short_description
-    except ValueError:
-        rank_zero_warn(f"Failed parsing docstring for {component}")
+    if component.__doc__ is None:
+        return None
+    if not _DOCSTRING_PARSER_AVAILABLE:
```

Comment:
qq:

`docstring_parse` is part of base `jsonargparse[signatures]`
https://github.com/omni-us/jsonargparse/blob/6dc85394e10ea4ee30e5f5dd2617bbee6e5c2893/setup.cfg#L23-L24

and isn't `jsonargparse[signatures]` required for cli?

Reply:
Yes. Though it is possible that someone installs pytorch-lightning without extras and installs jsonargparse without signatures.

Reply:
does cli work without signatures? I haven't used it much but when I did, it always recommended to install `jsonargparse[signatures]`. so just confirming this.

if it works with jsonargparse, what extensions does `signatures` provide?

Reply:
Yes, it works without `signatures`. Without it the docstrings are not used, which is quite a shortcoming because the help is not that useful. But the running of code is not affected. Some time ago `signatures` included more dependencies but were removed on some refactoring.

Reply:
then should we do `JSONARGPARSE_SIGNATURES_AVAILABLE` instead of `_DOCSTRING_PARSER_AVAILABLE`? since this is the only dependency within `signatures`.

Reply:
> then should we do `JSONARGPARSE_SIGNATURES_AVAILABLE` instead of `_DOCSTRING_PARSER_AVAILABLE`? since this is the only dependency within `signatures`.
> 
> just trying to keep only one external dependency here i.e `jsonargparse` and not users confuse it with another package that is not even present anywhere in PL requirements.

That does sound better to me. Later I can create a pull request with this change.

Reply:
Created #13035